### PR TITLE
fix(invite-members): add translations and better tests

### PR DESCRIPTION
**Pull request by ameliahsu**
fixes/cleanup for https://github.com/getsentry/sentry/pull/77528

- add translations
- better tests
- add missing types
- remove unnecessary `div`s

### CODE REVIEW

File: static/app/components/modals/inviteMembersModal/inviteMembersFooter.spec.tsx [hunk 1]
```diff
@@ -40,7 +43,7 @@ describe('InviteRowControlNew', function () {
       sendInvites: mockSendInvites,
     });
 
-    const sendButton = screen.getByLabelText(/send invite/i);
+    const sendButton = screen.getByRole('button', {name: 'Send invite'});
```
Comment (evanpurkhiser) [inline static/app/components/modals/inviteMembersModal/inviteMembersFooter.spec.tsx:46]: nice!

File: static/app/components/modals/inviteMembersModal/inviteMembersFooter.spec.tsx [hunk 2]
```diff
@@ -54,10 +57,11 @@ describe('InviteRowControlNew', function () {
...
         'moo-waan': {sent: true},
       },
-      willInvite: true,
     });
-    expect(screen.getByTestId('sent-invites')).toHaveTextContent(/2/i);
-    expect(screen.queryByTestId('failed-invites')).not.toBeInTheDocument();
+    expect(screen.getAllByRole('alert')).toHaveLength(1);
+    expect(
+      screen.getByRole('alert', {name: textWithMarkupMatcher('2 invites sent')})
+    ).toBeInTheDocument();
```
Comment (evanpurkhiser) [inline static/app/components/modals/inviteMembersModal/inviteMembersFooter.spec.tsx:64]: awesome!

File: static/app/components/modals/inviteMembersModal/inviteStatusMessage.tsx [hunk 3]
```diff
@@ -46,14 +41,18 @@ function CountMessage({sentCount, errorCount, isRequest}: CountMessageProps) {
     <div>
       {sentCount > 0 && (
         <StatusMessage status="success" isNewInviteModal>
-          <IconCheckmark size="sm" />
-          <span>{tct('[invites] sent.', tctComponents)}</span>
+          <IconCheckmark size="sm" color="successText" />
+          <span role="alert" aria-label={t('sent-invites')}>
```
Comment (evanpurkhiser) [inline static/app/components/modals/inviteMembersModal/inviteStatusMessage.tsx:45]:
```suggestion
          <span role="alert" aria-label={t('Sent Invites')}>
```

The aria label is human readable

File: static/app/components/modals/inviteMembersModal/inviteStatusMessage.tsx [hunk 4]
```diff
@@ -46,14 +41,18 @@ function CountMessage({sentCount, errorCount, isRequest}: CountMessageProps) {
...
+            {tct('[invites] sent.', tctComponents)}
+          </span>
         </StatusMessage>
       )}
       {errorCount > 0 && (
         <StatusMessage status="error" isNewInviteModal>
-          <IconWarning size="sm" />
-          <span>{tct('[failedInvites] failed to send.', tctComponents)}</span>
+          <IconWarning size="sm" color="errorText" />
+          <span role="alert" aria-label={t('failed-invites')}>
```
Comment (evanpurkhiser) [inline static/app/components/modals/inviteMembersModal/inviteStatusMessage.tsx:53]:
```suggestion
          <span role="alert" aria-label={t('Failed Invites')}>
```

File: static/app/components/modals/inviteMembersModal/inviteStatusMessage.tsx [hunk 5]
```diff
@@ -104,6 +103,7 @@ export default function InviteStatusMessage({
     if (isNewInviteModal) {
       return (
         <CountMessage
+          aria-label="alert"
```
Comment (evanpurkhiser) [inline static/app/components/modals/inviteMembersModal/inviteStatusMessage.tsx:106]: Was this one used?

Comment (ameliahsu) [reply static/app/components/modals/inviteMembersModal/inviteStatusMessage.tsx:106]: Ah nope, let me remove that

**Review by evanpurkhiser — APPROVED** [review]

**Comment by evanpurkhiser** [comment]
Great cleanup!!

**Pull request merged**